### PR TITLE
implements and logs carrying capacity

### PR DESCRIPTION
**Pull request by hadiyachishti**
implements and logs carrying capacity

### CODE REVIEW

File: carryingCap.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    with open(configFile, 'r') as f:
+        baseConfig = json.load(f)
+    baseConfig = baseConfig.get("sugarscapeOptions", baseConfig)
+
+    carryingCapacityReached = 0
+    extinctionOccurred = 0
+
+    for _ in range(numRuns):
+        config = baseConfig.copy()
+        config["seed"] = random.randint(0, 10000)
```
Comment (colinhanrahan) [inline carryingCap.py:35]: For extra security, we use `random.randint(0, sys.maxsize)` in other areas of the repo.

File: carryingCapPlot.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                populationLogs.append(json.load(f))
+
+    return populationLogs
+
+def plotPopulationOverTime(populationLogs):
+    carryingCapacities = []
+    
+    for log in populationLogs:
+        plt.plot(log, label=f'Seed {populationLogs.index(log) + 1}')
+        carryingCapacities.append(max(log))  # assuming the carrying capacity is the max population reached
```
Comment (colinhanrahan) [inline carryingCapPlot.py:20]: In `runSimulationAndLog` you consider the carrying capacity to be roughly the last ten values of the list. Maybe you can use the same logic here for runs that initially overshoot their carrying capacity?

File: sugarscape.py [hunk 3]
```diff
@@ -50,6 +50,11 @@ def __init__(self, configuration):
         self.activeQuadrants = self.findActiveQuadrants()
         self.configureAgents(configuration["startingAgents"])
         self.configureDiseases(configuration["startingDiseases"])
+        if configuration["headlessMode"] == False:
```
Comment (colinhanrahan) [inline sugarscape.py:53]: Looks like this is redundant with line 58 below?

**Comment by nkremerh** [comment]
This is ready for another look. @hadiyachishti please email me what the kernel here is for how we calculate carrying capacity. I think that belongs in `sugarscape.py` and should be part of updating the runtime stats (like Gini coefficient).

**Comment by nkremerh** [comment]
Resolved by 846d0a2.

**Pull request closed without merging**